x/group/server/invariant.go

```diff
@@ -0,0 +1,91 @@
+package server
+
+import (
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/regen-network/regen-ledger/x/group"
+)
+
+type resTally struct {
+	YesCount     int64
+	NoCount      int64
+	AbstainCount int64
+	VetoCount    int64
+}
+
+func resultVoteState(v group.Tally) resTally {
+	return resTally{
+		YesCount:     strToInt(v.YesCount),
+		NoCount:      strToInt(v.NoCount),
+		AbstainCount: strToInt(v.AbstainCount),
+		VetoCount:    strToInt(v.VetoCount),
+	}
+}
+
+func strToInt(str string) int64 {
+	integer, _ := strconv.ParseInt(str, 10, 64)
+	return integer
+}
+
+func (s serverImpl) RegisterInvariants(ir sdk.InvariantRegistry) {
+	ir.RegisterRoute(group.ModuleName, "Tally-Votes", s.TallyVotesInvariant())
+}
+
+func (s serverImpl) AllInvariants() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		res, stop := s.TallyVotesInvariant()(ctx)
+		if stop {
+			return res, stop
+		}
+		return sdk.FormatInvariant(group.ModuleName, "Tally-Votes", "\tTallyVoteSums is failed"), false
+	}
+}
+
+func (s serverImpl) TallyVotesInvariant() sdk.Invariant {
+	return func(sdkCtx sdk.Context) (string, bool) {
+		ctx := types.Context{Context: sdkCtx}
+		var pageReq = query.PageRequest{
+			Offset:     0,
+			Limit:      100,
```

Comment:
This means that we won't verify the proposal votes after 100 proposals, I guess we should just iterate through all the proposals using directly `PrefixScan(ctx, nil, nil)` and not use pagination at all here.